executor/runcexecutor/executor.go

```diff
@@ -171,7 +171,16 @@ func (w *runcExecutor) Run(ctx context.Context, id string, root executor.Mount,
 		bklog.G(ctx).Info("enabling HostNetworking")
 	}
 
-	resolvConf, err := oci.GetResolvConf(ctx, w.root, w.idmap, w.dns)
+	dns := w.dns
+	if meta.DNS != nil {
+		dns = &oci.DNSConfig{
+			Nameservers:   meta.DNS.Nameservers,
+			Options:       meta.DNS.Options,
+			SearchDomains: meta.DNS.Search,
+		}
+	}
+
+	resolvConf, err := oci.GetResolvConf(ctx, w.root, w.idmap, dns)
```

Comment:
It doesn't look like `GetResolvConf` is safe to be called like this. Looks like it does bunch of internal caching atm and can return a cached result for incorrect `dns` value.

Reply:
Oh, good catch, that makes things a bit trickier but I took a swing at it. Now it'll use a hashed filename derived from the DNS config.